units/commit_queue.go

```diff
@@ -0,0 +1,352 @@
+	QueueID  string `bson:"queue_id" json:"queue_id" yaml:"queue_id"`
+}
+
+func makeCommitQueueJob() *commitQueueJob {
+	job := &commitQueueJob{
+		Base: job.Base{
+			JobType: amboy.JobType{
+				Name:    commitQueueJobName,
+				Version: 0,
+			},
+		},
+	}
+	job.SetDependency(dependency.NewAlways())
+
+	return job
+}
+
+func NewCommitQueueJob(queueID string, id string) amboy.Job {
+	job := makeCommitQueueJob()
+	job.QueueID = queueID
+	job.SetID(fmt.Sprintf("%s:%s_%s", commitQueueJobName, queueID, id))
+
+	return job
+}
+
+func (j *commitQueueJob) Run(ctx context.Context) {
+	defer j.MarkComplete()
+
+	projectRef, err := model.FindOneProjectRef(j.QueueID)
+	if err != nil {
+		j.AddError(errors.Wrapf(err, "can't find project for queue id %s", j.QueueID))
+		return
+	}
+
+	if !projectRef.CommitQEnabled {
+		return
+	}
+
+	cq, err := commitqueue.FindOneId(j.QueueID)
+	if err != nil {
+		j.AddError(errors.Wrapf(err, "can't find commit queue for id %s", j.QueueID))
+	}
+	nextItem := cq.Next()
+	nextItemInt, err := strconv.Atoi(nextItem)
+	if err != nil {
+		j.AddError(errors.Wrapf(err, "can't parse next item \"%s\" as int", nextItem))
+		return
+	}
+
+	alreadyProcessed, err := versionExists(nextItemInt)
+	if err != nil {
+		j.AddError(errors.Wrap(err, "can't query for versions for this item"))
+	}
+	if alreadyProcessed {
+		return
+	}
+
+	githubToken, err := getGitHubToken()
+	if err != nil {
+		j.AddError(errors.Wrap(err, "can't get github token"))
+	}
+
+	pr, err := thirdparty.GetGithubPullRequest(ctx, githubToken, projectRef.Owner, projectRef.Repo, nextItemInt)
+	if err != nil {
+		j.AddError(errors.Wrap(err, "can't get PR from GitHub"))
+		return
+	}
+
+	if validatePR(pr) != nil {
+		j.AddError(errors.Wrap(err, "invalid PR"))
+		return
+	}
+
+	// GitHub hasn't yet tested if the PR is mergeable.
+	// Check back later
+	// See: https://developer.github.com/v3/pulls/#response-1
+	if pr.Mergeable == nil {
+		return
+	}
+
+	if !*pr.Mergeable {
+		err = sendCommitQueueGithubStatus(pr, message.GithubStateFailure, "PR not mergeable", "")
+		if err != nil {
+			j.AddError(errors.Wrap(err, "can't send github status"))
+		}
+		return
+	}
+
+	v, err := makeVersion(ctx, githubToken, projectRef, pr)
+	if err != nil {
+		j.AddError(errors.Wrap(err, "can't make version"))
+		return
+	}
+
+	err = sendCommitQueueGithubStatus(pr, message.GithubStatePending, "preparing to test merge", v.Id)
+	if err != nil {
+		j.AddError(errors.Wrap(err, "can't send github status"))
+	}
+
+	err = subscribeMerge(projectRef, pr, v.Id)
+	if err != nil {
+		j.AddError(errors.Wrap(err, "can't subscribe GitHub PR merge to version"))
+	}
+}
+
+func versionExists(prNumber int) (bool, error) {
+	patch, err := patch.FindOneByGithubPRNum(prNumber)
+	if err != nil {
+		return false, errors.Wrap(err, "can't query for patch matching PR Number")
+	}
+	return patch != nil, nil
+}
+
+func getGitHubToken() (string, error) {
+	conf, err := evergreen.GetConfig()
```

Comment:
Other jobs have an environment in a field in the job (private) that gets populated when nil, and then get the config from there.

This method, makes it difficult/hard to unittest/mock the config. I would try a different match.

Reply:
added env to job